src/backend/utils/fmgr/fmgr.c

```diff
@@ -789,7 +790,20 @@ fmgr_security_definer(PG_FUNCTION_ARGS)
 	if (set_sql_dialect && IsTransactionState())
 	{
 		if ((fcache->prolang == pltsql_lang_oid) || (fcache->prolang == pltsql_validator_oid))
+		{
 			sql_dialect_value = tsql_dialect;
+			if (set_local_schema_for_func_hook)
+			{
+				char 	*new_search_path = NULL;
+				new_search_path = (*set_local_schema_for_func_hook)(fcinfo->flinfo->fn_oid);
```

Comment:
this will not work, perf impact will be not acceptable. We are searching two catalogs and changing search path. This will create big penalties for function call. Let us see if we can optimize this by using fcache (store search path there). Also, do perf experiments to see how much penalty we hit due to this.